internal/message/watchutil.go

```diff
@@ -20,20 +45,36 @@ type Update[K comparable, V any] watchable.Update[K, V]
 // it handles the case where the watchable.Map already contains
 // entries before .Subscribe is called.
 func HandleSubscription[K comparable, V any](
+	meta Metadata,
 	subscription <-chan watchable.Snapshot[K, V],
-	handle func(Update[K, V]),
+	handle func(updateFunc Update[K, V], errChans chan error),
 ) {
+	errChans := make(chan error, 10)
```

Comment:
any reason for use 10 for buffer size?

Reply:
We need some buffers to make sure the errChan will not be stuck when observing errors from snapshot.Updates. As for the size, I am not sure about it, a suggested value from chatgpt.

Reply:
can you add some comments for it, or a `TODO: find a suitable value` for this.

Reply:
Done